### DISCUSSION TITLE
Tutorial/example project

### DISCUSSION THREAD

**Discussion by Lukasz-Juranek** [strictdoc-project/strictdoc#831] [Show and tell]
Hi, I've created some example project and tutorial for strictdoc
https://github.com/Lukasz-Juranek/AutomotiveDesignModel

https://youtu.be/k2MCFWvCs7E 

if you see it as valuable use it as you like,

**Comment by stanislaw** [comment]
Hi @Lukasz-Juranek,

Thanks a lot for sharing! I will make sure to watch it and will most probably include a link to your video in the documentation.

**Reply by stanislaw** [reply]
I have watched your video and found it very clearly explaining what you use StrictDoc for. In addition, being more of a space guy with the ECSS standards in the back of my mind, I have learned a thing or two about the SPICE 👍 

Here's some feedback along the lines that you presented:

- Could you change the Strictdoc to `StrictDoc` in the title of your video? `Automotive SPICE in opensource Strictdoc tool, with System Architecure ideas`
- StrictDoc is going to release the first UI MVP within 2-4 weeks. The shadow `server` command is already available but still has an experimental status.
- Reading a documentation tree with more than just a few trivial documents, indeed, takes some time and does not scale very well. There are a few optimizations that I have in mind, and they could significantly improve the incremental re-generation of the document tree. The spoiler: cache-pickle parts of the document tree graph and the traceability information and use it for making decisions for re-reading the whole thing over and over again or skipping some parts of it when they are known to not have to be changed. With this direction in mind, StrictDoc will resemble what compilers do when they decide how much should be recompiled/relinked.
- The support of VS Code and PyCharm could, indeed, be improved by setting up an Language Server Protocol (LSP) for the SDoc syntax and adding, like you suggested, a Table of Contents menu for navigating over the document's table of contents. Is this something you feel like you could contribute? 😃 
- You mentioned that many features are missing but you didn't specify anywhere, to my current knowledge, what they are. If you have a bunch of them, I encourage you to open a discussion ticket right here and provide them all as a single brain dump and then we will convert them into issues if needed. Otherwise, feel free to open a particular issue about the feature you are lacking the most. I would be curious to compare your nice/must-to-haves with what I and some other folks have in mind.

Thank you.
